src/NuGet.Core/NuGet.Packaging/Signing/TrustStore/X509StorePurpose.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Packaging.Signing
+{
+    public enum X509StorePurpose
```

Comment:
Does this need to be public?

Reply:
Alternatively, if you believe it should be kept public, from an API design point of view, should this enum be `[Flags]`?  Is there a possibility that an API caller might want to get an `X509Store` with certs with both/multiple purposes?

If not, do we need this enum, and instead should we get rid of `GetX509ChainFactory` and have one method for each purpose?

Reply:
The alternative is making this `internal`, creating a corresponding test enum type, and mapping from one to the other.